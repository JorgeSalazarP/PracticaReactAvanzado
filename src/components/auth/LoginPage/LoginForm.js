import React from 'react';
import T from 'prop-types';


const LoginForm = ({ onSubmit, setIsChecked }) => {

    const [credentials,setCredentials] = React.useState({        
        email: '',
        password: ''
    });
    const {email,password} = credentials;

    const handleChange = ev =>{

        setCredentials(oldCredentials => ({
            ...oldCredentials,
            [ev.target.name]: ev.target.value,
          }));

    }

    const handleSubmit = (ev) =>{
        ev.preventDefault();        
        onSubmit(credentials);
    }

    return (  

        <div className="container container_form ">

                <h1>Login in to Nodepop</h1>

                <form
                    onSubmit={handleSubmit}
                >
                    <div className="form_field">
                        <label htmlFor="email">Email</label>
                        <input 
                            type="email"
                            name="email"
                            className="loginForm-field"
                            value={email}
                            onChange={handleChange}
                        />
                    </div>

                    <div className="form_field">
                        <label htmlFor="password">Password</label>
                        <input 
                            type="password"
                            name="password"
                            className="loginForm-field"
                            value={password}
                            onChange={handleChange}
                        />
                    </div>
                    <div className="form_field">
                        <label htmlFor="checkbox">Remember password</label>
                        <input 
                            type="checkbox"
                            name="checkbox"
                            onChange={(ev)=>setIsChecked(ev.target.checked)}
                        />
                    </div>

                    <div className="form_field">
                        <input 
                            type="submit" 
                            className="btn btn-primary btn-lg btn-block" 
                            value="Login in" 
                            disabled={!email || !password}
                        />
                    </div>
                    
                </form>

                
        </div>


    );
}

LoginForm.propTypes = {
    setIsChecked: T.func.isRequired,
    onSubmit: T.func.isRequired,

};
export default LoginForm;