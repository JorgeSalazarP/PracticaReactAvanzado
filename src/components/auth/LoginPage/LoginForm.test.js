import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', ()=>{
    const props = {
        onSubmit: jest.fn(),
        setIsChecked: jest.fn()
    };
    const render = () => shallow(<LoginForm {...props}/>);

    test('should render',()=>{
        const wrapper = render();
        expect(wrapper.exists()).toBe(true);

    });

    test('should submit credentials',()=>{
        const credentials = { email: 'example@example.com', password: '1234'};
        const wrapper = render();
        const emailField = wrapper.find('.loginForm-field').at(0);
        emailField
        .props()
        .onChange({ target: {name: 'email', value: credentials.email }});

        const passwordField = wrapper.find('.loginForm-field').at(0);
        passwordField
        .props()
        .onChange({ target: {name: 'password', value: credentials.password }});

        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault:()=> {} });
        expect(props.onSubmit).toHaveBeenCalledWith(credentials);
    });

    

});


