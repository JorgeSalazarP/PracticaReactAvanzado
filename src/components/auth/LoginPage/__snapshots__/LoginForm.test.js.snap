// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should submit credentials 1`] = `
<div
  className="container container_form "
>
  <h1>
    Login in to Nodepop
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form_field"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="loginForm-field"
        name="email"
        onChange={[Function]}
        type="email"
        value="example@example.com"
      />
    </div>
    <div
      className="form_field"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="loginForm-field"
        name="password"
        onChange={[Function]}
        type="password"
        value="1234"
      />
    </div>
    <div
      className="form_field"
    >
      <label
        htmlFor="checkbox"
      >
        Remember password
      </label>
      <input
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="form_field"
    >
      <input
        className="btn btn-primary btn-lg btn-block loginSubmit"
        disabled={false}
        type="submit"
        value="Login in"
      />
    </div>
  </form>
</div>
`;

exports[`LoginForm snapshot testing 1`] = `
<div
  className="container container_form "
>
  <h1>
    Login in to Nodepop
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form_field"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        className="loginForm-field"
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form_field"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="loginForm-field"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form_field"
    >
      <label
        htmlFor="checkbox"
      >
        Remember password
      </label>
      <input
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="form_field"
    >
      <input
        className="btn btn-primary btn-lg btn-block loginSubmit"
        disabled={true}
        type="submit"
        value="Login in"
      />
    </div>
  </form>
</div>
`;
